### PR TITLE
Adds BBEdit to list of supported external Editors

### PR DESCRIPTION
**Pull request by NiklasBr**
Noticed that this app did not use the macOS default text editors for various file types, nor did not use the EDITOR in `.bash_profile`. Preferably Desktop should also tell us which editor is used in the context menu (e.g. Open with Foo rather than Open with Default Program), but that's for another time.

### CODE REVIEW

**Comment by niik** [comment]
Hey @NiklasBr, thanks for opening this. Looks like you've got a some code that's not formatted according to our prettifier standards. Please run `npm run prettify` and commit those changes.

**Comment by NiklasBr** [comment]
Thanks @niik , fixed and is working better now.

**Comment by niik** [comment]
Alright @NiklasBr, thanks for fixing those formatting issues. At this point there's a few assertions that aren't being met over on the Windows side of things. Since this editor is the first so far to be exclusive to macOS we're gonna need a little time to think through how to best deal with that. I'll ping @shiftkey here since he wrote a bunch of the win32 stuff and see if he has any good ideas.

**Comment by shiftkey** [comment]
@NiklasBr I had a chance to test your PR, and I've opened #3038 to address the underlying limitations (and better document things). When that's merged, all it should take is a merge into this branch (and maybe some cleanup) to integrate this work.

**Comment by NiklasBr** [comment]
Looks really good @shiftkey, the in #3038 docs are :italian chef kisses fingers:

**Comment by shiftkey** [comment]
@NiklasBr #3038 has been merged :tada:

File: app/src/models/editors.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export enum ExternalEditor {
   Atom = 'Atom',
   VisualStudioCode = 'Visual Studio Code',
   SublimeText = 'Sublime Text',
+  BBEdit = 'BBEdit',
```
Comment (j-f1) [inline app/src/models/editors.ts]: I think this can just be `BBEdit,`, unless you wrote it like this for clarity.

Comment (shiftkey) [reply app/src/models/editors.ts]: Sadly TypeScript wants them all to be initialized to a value:

Comment (j-f1) [reply app/src/models/editors.ts]: Oh, right 🤦‍♂️. If you don’t pass a value, it’s assumed to be a number.

**Comment by theconsultant** [comment]
Hey @NiklasBr which version of BBedit are you running? I've just updated to v.12 and the CLI binary has a different name and is found in a different location `/Contents/Helpers/bbedit_tool`.
P.s. thanks for setting this up.. I was about to try and figure it out and i'm glad you got to it first 👍

**Comment by NiklasBr** [comment]
@theconsultant I'm still running BBEdit 11, but I plan on purchasing the update soon.

A thing which makes me think my pull request could be better is that BBEdit registers `Contents/Helpers/bbedit_tool` at `/usr/local/bin/bbedit` and that maybe I should use that rather than a direct reference to the helper tool.

I need to think a bit about this.

**Comment by joshaber** [comment]
@NiklasBr @theconsultant Thank you both for your diligence on this PR. It's great to see 💖

**Comment by theconsultant** [comment]
I think you're right @NiklasBr that using /usr/local/bin/bbedit makes more sense.

I checked the manual, and BBEdit does offer to install the CLI tools on first launch, so it is safe to assume the tools are installed.

From the v12 manual: ```The first time you run BBEdit after installation, it will offer to install the “bbedit”, “bbdiff”, and “bbfind” command-line tools for you. If you choose not to do so, you can choose “Install Command-Line Tools” from the BBEdit (application) menu at any time to install (or re-install) the current version of each command line tool.```

**Comment by theconsultant** [comment]
@joshaber I'd rather fight with edge cases before deploy <grin>.

**Comment by j-f1** [comment]
@NiklasBr Can you fix the merge conflicts in this PR?

**Comment by NiklasBr** [comment]
I might have made some mistake merging and fixing the conflicts… Sorry

**Comment by j-f1** [comment]
@NiklasBr You can probably fix it with a bit of `git rebase -i` magic :)

**Comment by NiklasBr** [comment]
> git rebase -i

I never feel more stupid then trying to grok Git. I failed to make a proper rebase.

**Comment by guy-rouillier** [comment]
I'd like to see CodeLite added, and I imagine users of Code::Blocks would like that.  I appreciate that Visual Studio Code is present.  IDEs of various flavors are preferred for large projects.  Thanks.

**Pull request closed without merging**